invenio_app_ils/circulation/views.py

```diff
@@ -157,6 +199,48 @@ def post(self, **kwargs):
         return self.make_response(pid, loan, 202, links_factory=self.links_factory)
 
 
+class LoanSelfCheckoutResource(IlsCirculationResource):
+    """Loan self-checkout action resource."""
+
+    view_name = "loan_self_checkout"
+
+    @need_permissions("circulation-loan-self-checkout")
+    def get(self, **kwargs):
+        """Loan self-checkout GET method to retrieve items."""
+        try:
+            barcode = request.args["barcode"].upper()
+            assert barcode
+        except (KeyError, AssertionError):
+            msg = "Parameter `barcode` is missing"
+            raise MissingRequiredParameterError(description=msg)
+
+        try:
+            pid, item = self_checkout_get_item_by_barcode(barcode)
+        except ItemCannotCirculateError:
```

Comment:
The msg `Item not found ` is not very clear, as the user does not know what is an Item.
We should catch `ItemNotFound` here, and re-raise an exception with a different exc `LoanSelfCheckoutItemNotFound`, e.g. `Literature not found. Please try again with another barcode or contact the library.`

Reply:
Done ✅  and updated screeshots here: https://github.com/inveniosoftware/react-invenio-app-ils/pull/649